hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/TestDecommissioningStatus.java

```diff
@@ -307,6 +323,59 @@ public void testDecommissionStatus() throws Exception {
     AdminStatesBaseTest.cleanupFile(fileSys, file2);
   }
 
+  protected void verifyInitialState(FSNamesystem fsn, DatanodeManager dm)
+      throws InterruptedException {
+    dm.getDatanodes().forEach(datanodeDescriptor -> {
+      try {
+        checkDecommissionStatus(datanodeDescriptor, 0, 0, 0);
+      } catch (TimeoutException | InterruptedException e) {
+        throw new AssertionError("Datanode not in good state.", e);
+      }
+    });
+    int c = 0;
+    int totalBlocks;
+    long totalReplicatedBlocks;
+    while (true) {
+      totalBlocks = fsn.getBlockManager().getTotalBlocks();
+      totalReplicatedBlocks = fsn.getBlockManager().getTotalReplicatedBlocks();
+      if (totalBlocks == 4 && totalReplicatedBlocks == 4) {
+        break;
+      } else {
+        if (c == 4) {
+          throw new AssertionError("Unexpected Total blocks " + totalBlocks
+              + " and replicated blocks " + totalReplicatedBlocks);
+        }
+        Thread.sleep(3000);
+      }
+      c++;
+    }
```

Comment:
Can we use `GenericUtils.waitFor()`?

Reply:
Yeah I also thought about the same but then realized that `GenericUtils.waitFor()` would print thread stats as well if this fails. Since we are doing this validation to ensure the actual decommission test goes smooth, I thought keeping the error limited to just an `AssertionError` would be simpler, hence I did this way. I think this one also would be sufficient. If you still think `GenericUtils.waitFor()` must be used, I can change it. Thanks

Reply:
Also, keeping little complex logic of using counters (AtomicInteger as counters inside lambda) to ensure all 8 blocks are present in all DNs (4 blocks in each DN) would become even more complicated if we use `GenericUtils.waitFor()` so I think keeping this way is still less complex.

Reply:
Make sense to me. Thank you for the explanation.